bin/stager.py

```diff
@@ -0,0 +1,343 @@
+
+    def __init__(self):
+        self._setup_ssl_context()
+        self._parse_args()
+        if self.args.bundle:
+            self.stage_bundle(LocalBundle(self.args.bundle))
+        else:
+            logger.output(f"\nStaging bundles under \"{self.args.bundles}\":\n")
+            bundles = list(LocalBundle.bundles_under(self.args.bundles))
+            bundles.sort()
+            self.stage_bundles(bundles)
+        print("")
+
+    def _parse_args(self):
+        parser = argparse.ArgumentParser(description="Stage example bundles in S3.",
+                                         usage='%(prog)s [options]',
+                                         epilog="Default action is to stage all bundles under ./import/")
+        parser.add_argument('--target-bucket', metavar="<s3-bucket-name>", default=self.DEFAULT_BUCKET,
+                            help="stage files in this bucket")
+        parser.add_argument('--bundle', default=None, metavar="path/to/bundle",
+                            help="stage single bundle at this path")
+        parser.add_argument('--bundles', default='import', metavar="path",
+                            help="stage bundles under this path (must not include 'bundles')")
+        parser.add_argument('-q', '--quiet', action='store_true', default=False,
+                            help="silence is golden")
+        parser.add_argument('-t', '--terse', action='store_true', default=False,
+                            help="terse output, one character per file")
+        parser.add_argument('-l', '--log', default=None,
+                            help="log verbose output to this file")
+        parser.add_argument('-j', '--jobs', type=int, default=1,
+                            help="parallelize with this many jobs")
+        self.args = parser.parse_args()
+        if self.args.jobs > 1:
+            quiet = True
+            terse = True
+        else:
+            quiet = self.args.quiet
+            terse = self.args.terse
+
+        logger.configure(self.args.log, quiet=quiet, terse=terse)
+
+    def stage_bundles(self, bundles):
+        global executor
+        signal.signal(signal.SIGINT, self.signal_handler)
+        executor = ProcessPoolExecutor(max_workers=self.args.jobs)
+        self.total_bundles = len(bundles)
+        bundle_number = 0
+        for bundle in bundles:
+            bundle_number += 1
+            executor.submit(self.stage_bundle, bundle, bundle_number)
+        executor.shutdown()
+
+    def stage_bundle(self, bundle, bundle_number=None):
+        comment = f"({bundle_number}/{self.total_bundles})" if bundle_number else ""
+        signal.signal(signal.SIGINT, signal.SIG_DFL)
+        BundleStager(self.args.target_bucket).stage(bundle, comment)
+
+    @staticmethod
+    def signal_handler(signal, frame):
+        global executor
+        print('Shutting down...')
+        executor.shutdown()
+        sys.exit(0)
+
+    @staticmethod
+    def _setup_ssl_context():
```

Comment:
I ran into a problem at some point.  Jim's server, where are getting there datafiles, doesn't have an SSL cert.  On the other hand, the manifest files use http:// URLs.  I've turned it on for now and we'll see if it is a problem next time it runs.